### PR TITLE
Enable casting config values of most types

### PR DESCRIPTION
**Pull request by andrew4699**
# Description

Implementations of PolarisConfigurationStore may return strings regardless of the config type. In particular, one place where this happens when specifying a ConfigOverride in a test.

Currently, the interface checks the expected type for booleans and Lists and casts them to the correct config type. Whether this is a pattern we should proliferate or not is up for debate, but this PR simply extends the existing logic to work for other types, and it adds tests to ensure the different castings work. Without this PR, the test fails.

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

- [X] PolarisConfigurationStoreTest. Without this PR, the test fails.

# Checklist:

- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings

### CODE REVIEW

File: polaris-core/src/main/java/org/apache/polaris/core/PolarisConfigurationStore.java [hunk 1]
```diff
@@ -69,6 +69,16 @@ public interface PolarisConfigurationStore {
 
     if (config.defaultValue instanceof Boolean) {
       return config.cast(Boolean.valueOf(String.valueOf(value)));
+    } else if (config.defaultValue instanceof Short) {
```
Comment (eric-maynard) [inline polaris-core/src/main/java/org/apache/polaris/core/PolarisConfigurationStore.java:72]: I think we can leave off Short here. When would you make a config a short over an int?

File: polaris-core/src/main/java/org/apache/polaris/core/PolarisConfigurationStore.java [hunk 2]
```diff
@@ -69,6 +69,16 @@ public interface PolarisConfigurationStore {
...
+      return config.cast(Short.valueOf(String.valueOf(value)));
+    } else if (config.defaultValue instanceof Integer) {
+      return config.cast(Integer.valueOf(String.valueOf(value)));
+    } else if (config.defaultValue instanceof Long) {
+      return config.cast(Long.valueOf(String.valueOf(value)));
+    } else if (config.defaultValue instanceof Float) {
```
Comment (eric-maynard) [inline polaris-core/src/main/java/org/apache/polaris/core/PolarisConfigurationStore.java:78]: Similarly, I think we can leave off short. I don't know why you would want a config to be a float and not a double

File: polaris-core/src/main/java/org/apache/polaris/core/PolarisConfigurationStore.java (hunk 1 above)
Comment (andrew4699) [reply polaris-core/src/main/java/org/apache/polaris/core/PolarisConfigurationStore.java]: Updated

File: polaris-core/src/main/java/org/apache/polaris/core/PolarisConfigurationStore.java (hunk 2 above)
Comment (andrew4699) [reply polaris-core/src/main/java/org/apache/polaris/core/PolarisConfigurationStore.java]: Updated to remove float

File: polaris-core/src/test/java/org/apache/polaris/service/storage/PolarisConfigurationStoreTest.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import org.apache.polaris.core.PolarisConfiguration;
+import org.apache.polaris.core.PolarisConfigurationStore;
+import org.jetbrains.annotations.Nullable;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+/** Unit test for the default behaviors of the PolarisConfigurationStore interface. */
+public class PolarisConfigurationStoreTest {
+  @Test
+  public void testConfigsCanBeCastedFromString() {
```
Comment (eric-maynard) [inline polaris-core/src/test/java/org/apache/polaris/service/storage/PolarisConfigurationStoreTest.java:32]: Can we also add a test for the negative case? e.g. a string is provided for an int

Comment (andrew4699) [reply polaris-core/src/test/java/org/apache/polaris/service/storage/PolarisConfigurationStoreTest.java:32]: Added

**Review by eric-maynard — APPROVED** [review]
LGTM, looks good to merge
